pkg/virt-handler/virtwrap/schema.go

```diff
@@ -85,6 +108,7 @@ type Domain struct {
 
 type DomainStatus struct {
 	Status LifeCycle
+	Reason LifeCycle
```

Comment:
Maybe you can make another string subtype, something like `StateChangeReason`. It should be possible to merge the two reason types in there.